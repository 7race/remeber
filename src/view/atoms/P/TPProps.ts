export type TP = {
  children: string;
  type: 'h1' | 'h2' | 'small1' | 'small2';
};
