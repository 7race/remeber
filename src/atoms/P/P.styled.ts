import styled, { css } from 'styled-components';
import { TP } from './TPProps';

const pTypesTablet = {
  h1: css`
    font-size: 40px;
  `,
  h2: css`
    font-size: 20px;
  `,
  small1: css`
    font-size: 16px;
  `,
  small2: css`
    font-size: 14px;
  `,
};

const pTypesNotebook = {
  h1: css`
    font-size: 70px;
  `,
  h2: css`
    font-size: 50px;
  `,
  small1: css`
    font-size: 30px;
  `,
  small2: css`
    font-size: 20px;
  `,
};

export const P = styled.p<TP>`
  font-family: 'Tai Heritage Pro', serif;
  @media ${({ theme }) => theme.media.mobile}, ${({ theme }) => theme.media.tablet} {
    ${({ type }) => pTypesTablet[type]}
  }
  @media ${({ theme }) => theme.media.notebook} {
    ${({ type }) => pTypesNotebook[type]}
  }
`;
