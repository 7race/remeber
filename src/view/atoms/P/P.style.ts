import styled, { css } from 'styled-components';
import { TP } from './TPProps';

const pVariantsTablet = {
  h1: css`
    font-size: 40px;
  `,
  h2: css`
    font-size: 30px;
  `,
  small1: css`
    font-size: 16px;
  `,
  small2: css`
    font-size: 14px;
  `,
};

const pVariantsNotebook = {
  h1: css`
    font-size: 50px;
  `,
  h2: css`
    font-size: 50px;
  `,
  small1: css`
    font-size: 18px;
  `,
  small2: css`
    font-size: 16px;
  `,
};

export const P = styled.p<TP>`
  font-family: 'Tai Heritage Pro', serif;
  ${({ variant }) => pVariantsTablet[variant]}
  @media ${({ theme }) => theme.media.notebook} {
    ${({ variant }) => pVariantsNotebook[variant]}
  }
`;
