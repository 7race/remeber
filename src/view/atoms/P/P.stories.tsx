import { ComponentStory, ComponentMeta } from '@storybook/react';
import { P } from './P';

export default {
  title: 'P',
  component: P,
} as ComponentMeta<typeof P>;

const Template: ComponentStory<typeof P> = (args) => <P {...args} />;

export const Default = Template.bind({});

Default.args = {
  children: 'hello',
  variant: 'h1',
};
