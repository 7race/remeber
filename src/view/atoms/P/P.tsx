import { TP } from './TPProps';
import * as S from './P.style';

export const P = ({ children, type }: TP) => <S.P type={type}>{children}</S.P>;
