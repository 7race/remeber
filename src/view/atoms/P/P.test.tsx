import { render } from '@tests-utils';
import { P } from './P';

describe('P component', () => {
  it('correct styles: type=h1', () => {
    const { container } = render(<P variant='h1'>hello</P>);
    expect(container).toMatchSnapshot();
  });
  it('correct styles: type=h2', () => {
    const { container } = render(<P variant='h2'>hello</P>);
    expect(container).toMatchSnapshot();
  });
  it('correct styles: type=small1', () => {
    const { container } = render(<P variant='small1'>hello</P>);
    expect(container).toMatchSnapshot();
  });
  it('correct styles: type=small2', () => {
    const { container } = render(<P variant='small2'>hello</P>);
    expect(container).toMatchSnapshot();
  });
});
