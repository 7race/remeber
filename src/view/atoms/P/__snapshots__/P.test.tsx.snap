// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`P component correct styles: type=h1 1`] = `
<div>
  <p
    class="sc-bczRLJ kSucuR"
    type="h1"
  >
    hello
  </p>
</div>
`;

exports[`P component correct styles: type=h2 1`] = `
<div>
  <p
    class="sc-bczRLJ iwYdEx"
    type="h2"
  >
    hello
  </p>
</div>
`;

exports[`P component correct styles: type=small1 1`] = `
<div>
  <p
    class="sc-bczRLJ clFAbO"
    type="small1"
  >
    hello
  </p>
</div>
`;

exports[`P component correct styles: type=small2 1`] = `
<div>
  <p
    class="sc-bczRLJ cGGDap"
    type="small2"
  >
    hello
  </p>
</div>
`;
